android-json-form-wizard/src/main/java/com/vijay/jsonwizard/activities/JsonFormActivity.java

```diff
@@ -947,11 +945,41 @@ private void widgetWriteItemValue(String value, JSONObject item, String itemType
         if (!TextUtils.isEmpty(value)) {
             value = value.trim();
         }
-        item.put(JsonFormConstants.VALUE, itemType.equals(JsonFormConstants.HIDDEN) && TextUtils.isEmpty(value) ?
-                item.has(JsonFormConstants.VALUE) && !TextUtils.isEmpty(item.getString(JsonFormConstants.VALUE)) ?
-                        item.getString(JsonFormConstants.VALUE) : value : value);
+        if (itemType.equals(JsonFormConstants.HIDDEN) && TextUtils.isEmpty(value)) {
+            if (item.has(JsonFormConstants.VALUE) && !TextUtils.isEmpty(item.getString(JsonFormConstants.VALUE)))
+                item.put(JsonFormConstants.VALUE, item.getString(JsonFormConstants.VALUE));
+            else item.put(JsonFormConstants.VALUE, value);
+        } else {
+            if ((itemType.equals(JsonFormConstants.NATIVE_RADIO_BUTTON) || itemType.equals(JsonFormConstants.SPINNER) || itemType.equals(JsonFormConstants.CHECK_BOX)) && Utils.enabledProperty(NativeFormsProperties.KEY.WIDGET_VALUE_TRANSLATED)) {
+                if (itemType.equals(JsonFormConstants.SPINNER)) {
+                    if (item.has(JsonFormConstants.KEYS) && item.has(JsonFormConstants.VALUES)) {
+                        item.put(JsonFormConstants.VALUE, value);
+                    } else {
+                        item.put(JsonFormConstants.VALUE, Utils.generateTranslatableValue(value, item));
+                    }
+                } else {
+                    if (item.optString(JsonFormConstants.KEY).equalsIgnoreCase("reminders")) {
+                        item.put(JsonFormConstants.VALUE, value);
```

Comment:
@SebaMutuku This seems to be for a custom implementation. Should it be here? I think this should be moved to the Project that needs it.

Reply:
Done